File: contiguousSubarrayWithMaxSum.cpp
class Solution
{
public:
    int maxSubArray(vector<int> &arr)
    {
        int cur_sum, max_sum;
        cur_sum = max_sum = arr[0];
        for (int i = 1; i < arr.size(); i++)
        {
            cur_sum = max(arr[i], cur_sum + arr[i]);
            max_sum = max(cur_sum, max_sum);
        }
        return max_sum;
    }
};

File: contiguousSubarrayWithMaxSum.py
class Solution:
    def maxSubArraySum(self, arr: List[int]) -> int:
        cur_sum,max_sum=arr[0],arr[0]
        for i in range(1,len(arr)):
            cur_sum = max(arr[i],arr[i]+cur_sum)
            max_sum = max(cur_sum,max_sum)
        return max_sum